core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedClusterMessage.scala

```diff
@@ -49,6 +50,10 @@ private[spark] object CoarseGrainedClusterMessages {
   case class KillExecutorsOnHost(host: String)
     extends CoarseGrainedClusterMessage
 
+  case class UpdateExecutorsLogLevel(logLevel: String) extends CoarseGrainedClusterMessage
```

Comment:
Looks like this might have caused mima issues in Scala 2.13: https://github.com/apache/spark/pull/42358#pullrequestreview-1563933815

Reply:
If we need the mima check pass for both Scala 2.12 and Scala 2.13, `SparkAppConfig` might need to be designed as follows:

```scala
  case class SparkAppConfig(
      sparkProperties: Seq[(String, String)],
      ioEncryptionKey: Option[Array[Byte]],
      hadoopDelegationCreds: Option[Array[Byte]],
      resourceProfile: ResourceProfile)
    extends CoarseGrainedClusterMessage {

    private var _logLevel: Option[String] = None

    def logLevel: Option[String] = _logLevel

    def this(
        sparkProperties: Seq[(String, String)],
        ioEncryptionKey: Option[Array[Byte]],
        hadoopDelegationCreds: Option[Array[Byte]],
        resourceProfile: ResourceProfile,
        logLevel: Option[String]) = {
      this(sparkProperties, ioEncryptionKey, hadoopDelegationCreds, resourceProfile)
      this._logLevel = logLevel
    }
  }
```

Reply:
This is an internal class right? if so, ignoring it is OK

Reply:
OK, let's just ignore it

Reply:
Thanks for fixing this!